src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -285,6 +285,29 @@ bool CCARTree::train_machine(CFeatures* data)
 	return true;
 }
 
+void CCARTree::set_sorted_features(SGMatrix<float64_t>& sorted_feats, SGMatrix<int32_t>& sorted_indices)
+{
+	m_pre_sort=true;	
+	m_sorted_features=sorted_feats;
+	m_sorted_indices=sorted_indices;
+}
+
+void CCARTree::pre_sort_features(const SGMatrix<float64_t>& data, SGMatrix<float64_t>& sorted_feats, SGMatrix<int32_t>& sorted_indices)
+{
+	for(int32_t i=0; i<sorted_indices.num_cols; i++)
+		for(int32_t j=0; j<sorted_indices.num_rows; j++)
+			sorted_indices(j,i)=j;
```

Comment:
Couldn't find a way to memset this as we need to set rows to constant (not continuous in memory)